### ISSUE TITLE
(update) one thing; regexp

### ISSUE THREAD

**Issue by onetimesignup** [EchoDev/TwitchHTML5#10]
update; so the solution for me seems to be to completely skip the whole directory exclude, i'm now using just this line: `// @include /^https?://(www\.)?twitch\.tv/([0-9A-Z_a-z]+)?$/` which works perfectly on my end but only after restarting firefox, also in case you're wondering that line is indeed excluding vods.

-old-
update: WRONGGG, it has nothing to do with my regexp, script just seems to execute too fast or too slow when jumping from /directory -> streamer, i even used `/^.*twitch.*$/` to test it out and the script didnt execute correctly, im gonna experiment more with this, it may even turn out what i was told, that the waitForKeyElements must be tweaked for twitch

-old-
update; ooo I found a bug; being redirected from /directory to a stream prevents the script from launching in time due to the /directory part, i guess they use something like /directory?ref=streamname or something, gota figure this one out

update:  _these are the regexps you're looking for_

`// @include /^https?://(www\.)?twitch\.tv/([0-9A-Z_a-z]+)?$/`
`// @exclude /^https?://(www\.)?twitch\.tv/(directory|jobs|store)$/`

-old-
update: regexp isnt perfect, im gonna need help debugging it because i dont run into any issues with it
try this one: `// @include /^https?://(www\.)?twitch\.tv(/|/(?!directory|jobs|store)[0-9A-Z_a-z]+)$/`
It's not perfect but if you can't use this without the script crashing your browser then there's somethings bigger going on behind the scenes.

-old-
version: 1.4.1 (all the updates from 5 days ago)

1) "regexp doesn't exclude vods"
The regexp i provided you with is not 100%, it has to do with the /v/ exclude and wildcards but my brain is too nerfed figure it out atm

update: `// @include  /^https?://(www\.)?twitch\.tv/(?!(directory|jobs|store)$)([0-9A-Z\_a-z]+(/popout)?)?$/` <-- should give you what you want -->

* excluding vod-links & subpages ( directory/all )
* future support for '/popup' links in case you were to add it
* exclude non-live stream-pages like 'jobs' but allowing 'jobsxxx' named streams
* replaced the first wildcard to only check for "www" to avoid it from reacting on "tmi", "api" etc, I don't use mobile so i don't know if they have any special prefix, if so just add it.

2) "stalled player"
Twitch updated their player window recently forcing you to press the button for mature streams, if you only hide or remove that button/mask the player will now freeze and refuse to continue - in short; you either need to restore the box (unhide the button) or figure out a way to emulate the click / send accept data to the popup.

3) "script doesnt always load into the page forcing me to refresh the page"
When i experimented with twitch i also turned to `setTimeout` and `@run-at document-end` to try and have my script run correctly with twitchs ajax but i never ended up getting things to work seamlessly but I was recommended something that worked much better by being a lot less random; waitForKeyElements

https://gist.github.com/BrockA/2625891

Usage example:
`waitForKeyElements("div[class^='js-player']", html5);`
`function html5(){$("[class^='js-player']").html($("<iframe>").attr({........`

oh and you're gonna have to include these lines:
// @require     http://ajax.googleapis.com/ajax/libs/jquery/2.1.0/jquery.min.js
// @require     https://gist.github.com/raw/2625891/waitForKeyElements.js
// @grant       GM_addStyle

update: the `[class^='js-player']` switch allows the html5 player not only to play on the streamer-page (.js-player) but also to take over the front page (.js-player-container) and popout page (.js-player-container) if you one day decide to add support for them (main culprit `+$(location).attr('pathname')`).

I edited this post about a million times due to a broken a** sleepdeprived brai

**Comment by EchoDev** [comment]
Thanks for the info :)

I'll be adding number 1 and 3 into the script. I never had issue with number 2 tho. Have you hidden any other divs than in the script? Normally I just can click the button when a streamer sets the stream to matures only.

**Comment by EchoDev** [comment]
I think a stricter regex kills the script? When I include the new regex you provided the script stops working and needs a full refresh

using the old regex it works on every page. It might be better to just let the script run on twitch.tv/*

`// @include /^https?://(.*\.)?twitch\.tv/.*(?!/v/).*$/`

**Comment by onetimesignup** [comment]
1) "Unresponsive script"
I ran into this yesterday when I was messing with the script but couldn't figure out where it came from.
Firefox reported it being an issue with the html5 page and altering the regexp didn't help for me and then this morning it just stopped screwing with me.
regexp type1: `/^https?://(www\.)?twitch\.tv/(?!(directory|jobs|store)$)([0-9A-Z_a-z]+(/popout)?)$/` 99% same as the one I initially posted, only removed the escaping of the underline/score as it wasn't needed
regexp type2: `/^https?://(www\.)?twitch\.tv/([0-9A-Z_a-z]+)?$/` less complicated regexp that will exclude vods but also ignore the popup-window

2) "I never had issue with number 2"
It might be a issue with latest firefox (v44) but if you allow the script to delete/hide the button rather than pressing it when you press 'play' on the html5 player it instantly stops, if you hold space it will brute-play the steam but heh, no, you don't want to do it that way.

If it's not running the latest firefox + greasemonkey then it might be related to browser cache, I mean I always empty mine when I close the browser and I close it like 10+ times a day because I don't like keeping anything open so when I return to streamers that has the mature content option enabled I have to re-click the button.

X) "// @grant    GM_addStyle"
You're gonna want to use this because they changed something (access rights?) in later versions of greasemonkey, if I don't have that line in my script I will be stuck at a twitch logo page.

**Comment by EchoDev** [comment]
1) regex
Anytime I add a complicated version of regex it seems to kill the script for me, it forces me to do a full refresh. Having a simple regex makes it run much more reliable for me. 

2) mature streams
Do you have an example of mature streams for me to test? I let Firefox cache data to RAM so I also always have a clean cache but never seen the button. (tho it is probably stored in cookie, I should check that)

3) "// @grant GM_addStyle"
I'm running Firefox 44 with Greasemonkey 3.6
I don't have that line in my script but the script works fine for me. What version of Greasemonkey are you using?

**Comment by onetimesignup** [comment]
1) Try switching to type2, if that doesn't work with you I may have to come up with some crazy ass voodoo magic.

2)  hold on, gimme a second here

3) Greasemonkey 3.6 - issue sauce: http://stackoverflow.com/a/34298083

**Comment by EchoDev** [comment]
1) I'm not sure what to do with the regex yet. Entering/exiting vods might break the script again. I might just let the script run on the full domain

2) I'm not seeing any buttons or anything atm, I will look into it later today

3) Yeah I was guessing it was because I don't include jquery into the script. I think it is not necessary because it loads jquery from Twitch itself.

**Comment by EchoDev** [comment]
So I just fiddled around with the mature button. For me this happens:

1. I open a Twitch link that has the stream set to mature in a private window (so I have no cookies or cache).
2. I click on the "Start Watching button"
3. I click on the play button, stream starts playing

It looks to work fine for me. Did you try running the script without including jquery in the script? It might mess some things up.

**Comment by onetimesignup** [comment]
Yea well I mean if you don't hide the "start watching" button then there's no problem watching the stream, it was when you used the .hide to remove that window that the player refused to start when you pressed play- I think you removed the .hide() stuff in the latest/later update

**Comment by EchoDev** [comment]
The script should match the full domain, if a user enters a page that would be excluded (eg /directory) it won't catch the AJAX-requests and thus breaking the script and needing a refresh. So this can be closed.
